src/lib/storage.ts

```diff
@@ -88,6 +95,31 @@ export class Storage implements PeprStore {
     this.#dispatchUpdate("add", [key], value);
   };
 
+  /**
+   * Creates a promise and subscribes to the store, the promise resolves when
+   * the key and value are seen in the store.
+   *
+   * @param key - The key to add into the store
+   * @param value - The value of the key
+   * @returns
+   */
+  setItemAndWait = (key: string, value: string) => {
+    this.#dispatchUpdate("add", [key], value);
+    return new Promise<void>((resolve, reject) => {
+      const unsubscribe = this.subscribe(data => {
+        if (data[key] === value) {
+          unsubscribe();
+          resolve();
+        }
+        // If promise has not resolved before MAX_WAIT_TIME reject
+        setTimeout(() => {
+          unsubscribe();
+          reject();
+        }, MAX_WAIT_TIME);
+      });
```

Comment:
```suggestion
      const unsubscribe = this.subscribe(data => {
        if (data[key] === value) {
          unsubscribe();
          resolve();
        }
      });
      
      // If promise has not resolved before MAX_WAIT_TIME reject
      setTimeout(() => {
        unsubscribe();
        reject();
      }, MAX_WAIT_TIME);
```

Not 100% sure here on the syntax (cuz the PR suggestion editor's kinda not-great) but it seems like you'd want to establish the setTimeout() just the one time -- right after you subscribe to the store event -- rather than _every_ time the subscribe handler is called.  No?  :thinking:

Reply:
yes, good catch